autogluon/searcher/searcher.py

```diff
@@ -155,25 +156,19 @@ def get_config(self, **kwargs):
         -------
         A new configuration that is valid.
         """
-        if len(self._results) == 0: # no hyperparams have been tried yet, first try default config
+        if not self._results:  # no hyperparams have been tried yet, first try default config
             new_config = self.configspace.get_default_configuration().get_dictionary()
         else:
             new_config = self.configspace.sample_configuration().get_dictionary()
         with self.LOCK:
             num_tries = 1
             while pickle.dumps(new_config) in self._results.keys():
                 assert num_tries <= self.MAX_RETRIES, \
-                    "Cannot find new config in BaseSearcher, even after {} trials".format(
-                        self.MAX_RETRIES)
+                    f"Cannot find new config in BaseSearcher, even after {self.MAX_RETRIES} trials"
                 new_config = self.configspace.sample_configuration().get_dictionary()
                 num_tries += 1
-            self._results[pickle.dumps(new_config)] = 0
+            self._results[pickle.dumps(new_config)] = float("-inf")
```

Comment:
Could you give an example where this alters behavior from the existing code? Are certain eval metrics defective at present because of this?